internal/webhooks/cluster.go

```diff
@@ -343,34 +352,18 @@ func (webhook *Cluster) validateTopology(ctx context.Context, oldCluster, newClu
 				field.Invalid(
 					fldPath.Child("version"),
 					oldCluster.Spec.Topology.Version,
-					fmt.Sprintf("old version %q cannot be compared with %q", oldVersion, inVersion),
+					"old version must be a valid semantic version",
 				),
 			)
 		}
-		if inVersion.NE(semver.Version{}) && oldVersion.NE(semver.Version{}) && version.Compare(inVersion, oldVersion, version.WithBuildTags()) == -1 {
-			allErrs = append(
-				allErrs,
-				field.Invalid(
-					fldPath.Child("version"),
-					newCluster.Spec.Topology.Version,
-					fmt.Sprintf("version cannot be decreased from %q to %q", oldVersion, inVersion),
-				),
-			)
-		}
-		// A +2 minor version upgrade is not allowed.
-		ceilVersion := semver.Version{
-			Major: oldVersion.Major,
-			Minor: oldVersion.Minor + 2,
-			Patch: 0,
-		}
-		if inVersion.GTE(ceilVersion) {
-			allErrs = append(
-				allErrs,
-				field.Forbidden(
-					fldPath.Child("version"),
-					fmt.Sprintf("version cannot be increased from %q to %q", oldVersion, inVersion),
-				),
-			)
+
+		if _, ok := newCluster.GetAnnotations()[clusterv1.ClusterTopologyUnsafeUpdateVersionAnnotation]; ok {
+			log := ctrl.LoggerFrom(ctx)
+			log.Info(fmt.Sprintf("Skipping version validation for object because annotation %q is set", clusterv1.ClusterTopologyUnsafeUpdateVersionAnnotation))
```

Comment:
@sbueringer should we add a warning to allWarnings here?

Reply:
Yup, good idea (additionally)